### PR TITLE
feat(reporting-rate-summary): working get with filters and dimensions

### PR DESCRIPTION
**Pull request by HendrikThePendric**
This makes the call for getting reporting-rate-summaries work. It only addresses the api side of things. Further updates in in the components/actions/reducers are still needed, but I have understood from you (@Mohammer5 ) that you have a branch ready in which you have implemented these changes already. Let's merge and see where we are.

Also, we need to create/update a component that renders the data. But that's for another branch....

### CODE REVIEW

File: src/redux/actions/organisationUnits.js [hunk 1]
```diff
@@ -40,9 +40,12 @@ export const loadOrganisationUnits = () => dispatch =>
             )
         })
 
-export const selectOrganisationUnit = (_event, { id, path, displayName }) => ({
+export const selectOrganisationUnit = (
+    _event,
```
Comment (Mohammer5) [inline src/redux/actions/organisationUnits.js]: I think the `_event` parameter should be removed from this function, the mapDispatchToProps function should handle this (why should a function accept a parameter it doesn't use)

File: src/utils/api.js [hunk 2]
```diff
@@ -130,31 +131,55 @@ export const postDataSetReportComment = (
...
+    }
+
+    return d2.analytics.aggregate.get(req)
+}
+
+/**
+ * @param {Object} orgUnit
+ * @returns {Promise} - Array of IDs of the orgUnit and its direct descendants
+ */
+export const getOrgUnitAndChildrenIds = orgUnit => {
```
Comment (Mohammer5) [inline src/utils/api.js]: This function contains duplicate code. What about rewriting it like this:
```js
export const getOrgUnitAndChildrenIds = orgUnit => {
    const children = orgUnit.children.size
        ? Promise.resolve(orgUnit.children)
        : d2.models.organisationUnits
            .get(orgUnit.id, { fields: ['children[id]'] })
            .then(({ children }) => children)

    return children
        .then(children => [
            orgUnitWithChildren.id,
            ...mapResponseToArrayOfIds(children)
        ]);
}
```

File: src/redux/actions/organisationUnits.js (hunk 1 above)
Comment (HendrikThePendric) [reply src/redux/actions/organisationUnits.js]: Yes, you are right. I spotted before what you were doing in the `mapStateToProps` functions, and that is indeed much more elegant. But that would be a change for when we are working on the component/action/reducer. And I will do so once your work has been merged.

File: src/utils/api.js (hunk 2 above)
Comment (HendrikThePendric) [reply src/utils/api.js]: Nice one 👍

**Pull request merged**